### DISCUSSION TITLE
How to send etherium along with a transaction ?

### DISCUSSION THREAD

**Discussion by Navin77773** [Cyfrin/foundry-full-course-cu#337] [Q&A] [answered]
I want to send etherium along with a transaction by calling pay function in below code . After when i check the balance of the address it shows 0
code -
```solidity
//SPDX-License-Identifier:MIT
pragma solidity ^ 0.8.17;

contract PayandWithdraw{
    address[] funders;
    mapping(address => uint) addresstoeth;

    address owner;

    constructor(){
        owner = payable(msg.sender);
    }

    error OnlyOwner();
    modifier OnlyOwnercancall{
        if(msg.sender != owner) revert OnlyOwner();
        _;
    }

    function pay() payable public {
        funders.push(payable(msg.sender));
        addresstoeth[payable(msg.sender)]=msg.value;
    }
    function checkBalanceforOwner(address passaddress) private view returns(uint){
        return addresstoeth[passaddress];
    }
    function checkUserBalance() public view returns(uint){
        return addresstoeth[msg.sender];
    }

    function withdraw() private OnlyOwnercancall() {
        for (uint i;i<=funders.length;i++){
            address temp = funders[i];
            addresstoeth[temp]=0;
        }
        funders = new address[](0);
        payable(msg.sender).transfer(address(this).balance);
    }
}'''
```

linux command : cast send 0x5FbDB2315678afecb367f032d93F642
f64180aa3 "pay()" 1e18 --rpc-url $RPC_URL --private-key $PRIVATE_KEY
output : cast call 0x5FbDB2315678afecb367f032d93F642f64180aa3 "checkUserBalance()"
0x0000000000000000000000000000000000000000000000000000000000000000

**Comment by shikhar229169** [comment, marked as answer]
The correct way is:
```
cast send 0x5FbDB2315678afecb367f032d93F642f64180aa3 "pay()" --value 1ether --rpc-url $RPC_URL --private-key $PRIVATE_KEY
```

Also you can refer to the docs:
```
https://book.getfoundry.sh/reference/cast/cast-send?highlight=cast%20send#cast-send
```

**Reply by Navin77773** [reply]
still getting same answer ,
cast send  0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0 "pay()" --value 1ether -
-rpc-url $RPC_URL --private-key $PRIVATE_KEY
cast call 0x5FbDB2315678afecb367f032d93F642f64180aa3 "checkUserBalance()"
output : - 0x

**Reply by shikhar229169** [reply]
This means that the contract is not deployed. 
When you want to deploy your contract on a network you need to add --broadcast at the end of the deploy command, otherwise only simulation will take place.

**Reply by Navin77773** [reply]
No , i have actually deployed it and used broadcast command . Thanks for you responce by the way

**Reply by shikhar229169** [reply]
Earlier you were getting 0x0000000000000000000000000000000000000000000000000000000000000000, as an output that signifies that it is actually being read from a contract that is deployed.

But now you are getting only 0x, that signifies that the contract is not deployed

**Reply by shikhar229169** [reply]
Oh ok, I got it. 
while calling send you are passing contract - 0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0
but while doing cast call you are passing contract - 0x5FbDB2315678afecb367f032d93F642f64180aa3

**Reply by cromewar** [reply]
Does @shikhar229169 solution worked for you @Navin77773 ?

**Comment by Navin77773** [comment]
@cromewar  @shikhar229169  i recompiled but still i get same answer 
input : - 
cast send 0x5FbDB2315678afecb367f032d93F642f64180aa3 "pay()" --value 3ether --rpc-url $RPC_URL --private-key $PRIVATE_KEY
cast call  0x5FbDB2315678afecb367f032d93F642f64180aa3 "checkUserBalance()" 
output : - 
0x0000000000000000000000000000000000000000000000000000000000000000

cast --to-base 0x0000000000000000000000000000000000000000000000000000000000000000 dec
0

**Reply by shikhar229169** [reply]
I think there is something wrong with your contract. Consider pasting it here.

**Reply by Navin77773** [reply]
Got the answer the script code was wrong

**Comment by 1nikkas** [comment]
I will leave a comment here to follow the solution of the problem
